src/components/NewAnalysisModal.js

```diff
@@ -24,33 +24,41 @@ export const NewAnalysisModal = Utils.withDisplayName('NewAnalysisModal')(
     const [busy, setBusy] = useState()
 
     const renderNewRuntimeModal = tool => h(NewRuntimeModalBase, {
-      isOpen: true, //always true, this is detached from dom for other viewmodes within this modal
+      isOpen: viewMode === NEW_JUPYTER_MODE || viewMode === NEW_RSTUDIO_MODE,
```

Comment:
this was actually a bug. @rtitle, you actually identified it when you asked if it should re-open to the previous page. I thought it was just remembering choices, but in fact it was still attached because the viewMode was never reset on modal close, which is a bit annoying I found after some playing around with it. Same thing was happening with new modal